json-endpoints/jetpack/class.jetpack-json-api-plugins-install-endpoint.php

```diff
@@ -78,12 +77,20 @@ protected static function get_plugin_id_by_slug( $slug ) {
 			return false;
 		}
 		foreach( $plugins as $id => $plugin_data ) {
-			if ( strpos( $id, $slug ) !== false ) {
+			if ( self::get_slug_from_id( $id ) === $slug ) {
 				return $id;
 			}
 		}
 		return false;
 	}
+
+	protected static function get_slug_from_id( $id ) {
+		$dir = dirname ( $id );
+		if ( $dir === '.' ) {
```

Comment:
Looks like we are already doing something simular to what I have above see 
https://github.com/Automattic/jetpack/blob/master/json-endpoints/jetpack/class.jetpack-json-api-plugins-endpoint.php#L191